apps/studio/src/server/modules/page/__tests__/page.router.test.ts

```diff
@@ -0,0 +1,1206 @@
+        }),
+      )
+    })
+
+    it("should throw 400 if folderId is not a Folder resource", async () => {
+      // Arrange
+      const { site, page } = await setupPageResource({ resourceType: "Page" })
+      const expectedPageArgs = {
+        siteId: site.id,
+        title: "Test Page",
+        permalink: "test-page",
+        folderId: Number(page.id),
+      }
+
+      // Act
+      const result = caller.createPage({
+        ...expectedPageArgs,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "BAD_REQUEST",
+          message: "Folder not found or folderId is not a folder",
+        }),
+      )
+    })
+
+    // TODO: Implement tests when permissions are implemented
+    it.skip("should throw 403 if user does not have access to site", async () => {})
+
+    it.skip("should throw 403 if user does not have write access to folder", async () => {})
+
+    it.skip("should throw 403 if user does not have write access to root", async () => {})
+  })
+
+  describe("getRootPage", () => {
+    it("should throw 401 if not logged in", async () => {
+      const unauthedSession = applySession()
+      const unauthedCaller = createCaller(createMockRequest(unauthedSession))
+
+      const result = unauthedCaller.getRootPage({
+        siteId: 1,
+      })
+
+      await expect(result).rejects.toThrowError(
+        new TRPCError({ code: "UNAUTHORIZED" }),
+      )
+    })
+
+    it("should return 404 if site does not exist", async () => {
+      // Act
+      const result = caller.getRootPage({
+        siteId: 999999, // should not exist
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message: "Root page not found",
+        }),
+      )
+    })
+
+    it("should return the root page successfully", async () => {
+      // Arrange
+      const { site, page } = await setupPageResource({
+        resourceType: "RootPage",
+      })
+
+      // Act
+      const result = await caller.getRootPage({
+        siteId: site.id,
+      })
+
+      // Assert
+      expect(result).toMatchObject(pick(page, ["id", "title", "draftBlobId"]))
+    })
+
+    it("should return 404 if root page does not exist", async () => {
+      // Arrange
+      const { site } = await setupSite()
+
+      // Act
+      const result = caller.getRootPage({
+        siteId: site.id,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message: "Root page not found",
+        }),
+      )
+    })
+
+    it.skip("should throw 403 if user does not have access to site", async () => {})
+
+    it.skip("should throw 403 if user does not have read access to root", async () => {})
+  })
+
+  // TODO: Implement tests when publish works
+  describe.skip("publishPage", () => {})
```

Comment:
@harishv7 - i think it'd be good for us to write **test descriptors**, wdyt? can you write skipped tests for behaviour that's important to verify here?

Reply:
https://github.com/opengovsg/isomer/pull/770